dockstore-common/src/main/java/io/dockstore/common/yaml/DockstoreYamlHelper.java

```diff
@@ -373,40 +382,82 @@ private static void addNewPropertyClassToQueue(String dockstoreYamlPackageName,
+                    .map(v -> buildMessageFromViolation(v, validateeDescription))  // NOSONAR a lambda is more understandable than method reference here
                     .collect(Collectors.joining("; ")));
         }
     }
 
-    private static <T> String buildMessageFromViolation(ConstraintViolation<T> violation) {
-        String message = violation.getMessage();
-
-        // If the violation contains a non-empty property path, add it to the message, with a summary of the invalid value, if present.
+    /**
+     * Creates an error message from a constraint violation, using a description of the
+     * validatee (object being validated) if the property is not available.
+     * @param violation constrain violation to convert
+     * @param validateeDescription string that described the object being validated (ex: ".dockstore.yml")
+     * @return a well-formed error message
+     */
+    private static <T> String buildMessageFromViolation(ConstraintViolation<T> violation, String validateeDescription) {
+        // Determine the subject of the error message, either a property name or the validatee description.
+        String subject = null;
         javax.validation.Path propertyPath = violation.getPropertyPath();
         if (propertyPath != null) {
             String propertyPathString = propertyPath.toString();
             if (propertyPathString != null && !propertyPathString.isEmpty()) {
-                message = String.format("Property \"%s\" %s", propertyPathString, message);
-                Object invalidValue = violation.getInvalidValue();
-                if (invalidValue != null) {
-                    message = String.format("%s (current value: \"%s\")", message, StringUtils.abbreviate(invalidValue.toString(), INVALID_VALUE_ECHO_LIMIT));
-                }
+                subject = String.format("Property \"%s\"", propertyPathString);
             }
         }
+        if (subject == null) {
+            subject = validateeDescription;
+        }
+
+        // Create the message by prepending the subject to the verb phrase in the violation message.
+        String message = String.format("%s %s", subject, violation.getMessage());
+
+        // If there's a non-null invalid value, and it's a simple object (not the invalid bean), include it in the message.
+        Object invalidValue = violation.getInvalidValue();
+        if (invalidValue != null && violation.getLeafBean() != invalidValue) {
+            message = String.format("%s (current value: \"%s\")", message, StringUtils.abbreviate(invalidValue.toString(), INVALID_VALUE_ECHO_LIMIT));
+        }
 
         return message;
     }
 
+    /**
+     * Determine if a violation references a property in .dockstore.yml that represents an entry (tool/workflow/etc).
+     * @param violation
+     */
+    private static <T> boolean doesNotReferenceWorkflowish(ConstraintViolation<T> violation) {
```

Comment:
Following your JavaDoc, you could rename this method to `doesNotReferenceEntry`.

Reply:
Done.